overlord/overlord.go

```diff
@@ -402,17 +402,33 @@ func (o *Overlord) requestRestart(t state.RestartType) {
 	}
 }
 
+var preseedExitWithError = func(err error) {
+	fmt.Fprintf(os.Stderr, "cannot preseed: %v\n", err)
+	os.Exit(1)
```

Comment:
Because the task runner runs concurrently and this is exiting the entire process, is there any risk that some of the operations performed by other goroutines is in a state where it should not be interrupted? I'm mainly asking if this should be changed into a graceful shutdown instead of an abrupt exit.

Reply:
Not during preseeding where we run a well defined sequence of seeding tasks.